### ISSUE TITLE
[Table] real-time data is lost when sorting

### ISSUE THREAD

**Issue by larkin** [nasa/openmct#1738]
When viewing a telemetry table that has both historical and realtime data, if you sort the table, then all realtime data will be removed during the sort, so that only the historical data still persists.  Sorting thus results in gaps appearing in the data.

Easy to reproduce on master by looking at a sinewave in table mode, setting time conductor to follow mode with a ten second window, and then resorting.

![table-realtime-sort-data-loss](https://user-images.githubusercontent.com/226503/30982171-748c0a54-a43b-11e7-84df-27a8d231ddb3.gif)

If there's a simple fix, this should be high priority.

**Comment by charlesh88** [comment]
Was seeing something that might be related to this, in the plot: switching from plot view to table, then back to plot, throws out the data while the table was loading and draws a linear connector line where the plot left off and where it picks up again:

![screen shot 2017-09-28 at 11 02 44 am](https://user-images.githubusercontent.com/1056412/30982609-bc61fa72-a43c-11e7-9862-bb5c11679988.png)
![screen shot 2017-09-28 at 11 02 53 am](https://user-images.githubusercontent.com/1056412/30982614-c097f772-a43c-11e7-9392-325f38e43486.png)
![screen shot 2017-09-28 at 11 03 12 am](https://user-images.githubusercontent.com/1056412/30982617-c45cbb0e-a43c-11e7-9632-3e284ba798e0.png)

@larkin If this is a separate issue, let me know and I'll file it.

**Comment by VWoeltjen** [comment]
Is this reproducible with anything other than SWGs? (Could it be an SWG bug?)

**Comment by larkin** [comment]
@charlesh88 @VWoeltjen that's old plot library, I'd ignore it for now unless it comes up in new plot.

**Comment by VWoeltjen** [comment]
Candidate fix on `table-rows-1738`. It's not clear whether it makes more sense to have the responsibility shown in that commit in TelemetryTableController or MCTTableController. The communication model as a whole for those components seems more complex than necessary, and presents some opportunities for both user code and the table itself to retain redundant state.

**Comment by akhenry** [comment]
> The communication model as a whole for those components seems more complex than necessary, and presents some opportunities for both user code and the table itself to retain redundant state

It totally, totally is, and I can only apologize profusely for that, it's entirely my fault. The plan was always to eventually remove Angular anyway, which will simplify communications between the components. Another aspect of the complication in this component is the (probably unnecessary) desire to retain MCTTable as a generic 'table' component, without hardwiring too much telemetry table specific stuff into it. As a consequence there's some abstraction and generalization between the components that just makes the whole thing a horrible mess. I'd be in favor of doing away with MCTTable as a generic component if it simplifies things.

In short, MCTTable needs to be rewritten, without Angular involved. It currently has a very awkward relationship with Angular anyway due to performance concerns).

**Comment by larkin** [comment]
@VWoeltjen can we open a PR for your candidate fix?  I'd like to get this merged in.

**Comment by dtailor90** [comment]
Testathon 2/21/18: Verified Fixed with new plots
